lib/iterator.js

```diff
@@ -199,6 +199,67 @@ class Iterator {
     return result + suffix;
   }
 
+  // Find value in this iterator compared by `comparator` on values
+  // obtained by applying `accessor` to each value.
+  // Signature: comparator, [accessor[, thisArg]]
+  //   comparator <Function> returns true if new value should be accepted.
+  //     currValue <any> current value, starts with undefined
+  //     nextValue <any> next value
+  //     Returns: <boolean> true if next value should be accepted.
+  //   accessor <Function> gets value to compare by, current iterator value
+  //       is used by default.
+  //     value <any> this
+  //     Returns: <any> value to compare by.
+  //   thisArg <any> value to be used as `this` when calling `accessor` and
+  //       `comparator`.
+  // Returns: element with maximum `accessor` value.
```

Comment:
It's unclear what 'maximum' value is for complex types, besides that, you can even use this function to find a minimum value, as you do later in this file. You also need to specify that it will return undefined when the value couldn't be found.